platform/src/main/java/org/hillview/table/filters/StringFilterFactory.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.hillview.table.filters;
+
+import org.apache.commons.lang.StringUtils;
+import org.hillview.table.api.IStringFilter;
+
+import javax.annotation.Nullable;
+import java.util.regex.Pattern;
+
+public class StringFilterFactory {
+    public static IStringFilter instance = null;
+
+    public IStringFilter getFilter(StringFilterDescription stringFilterDescription) {
```

Comment:
This function was fine previously. You should revert to the previous implementation.